src/Chapter4.hs

```diff
@@ -629,7 +659,16 @@ Can you implement a monad version of AND, polymorphic over any monad?
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+andM = undefined
+-- I didnt understand how to solve this task. I tried these:
+-- andM mx my = mx >>= (\x -> my >>= (\y -> pure (x && y)))
+-- andM mx my = mx >>= (\x -> fmap (&& x) my )
+-- andM mx my = mx >>= (\x -> pure (&& x) <*> my )
+-- but they all failed 1 test:
+-- it "Just False - Nothing" $ andM (Just False) Nothing `shouldBe` Just False
+-- to be honest i dont understand why it should be Just False in this case but at the same time it should be Nothing in these cases:
+-- it "Nothing - Just" $ andM Nothing (Just True) `shouldBe` Nothing
+-- it "Just True - Nothing" $ andM (Just True) Nothing `shouldBe` Nothing
```

Comment:
The ordinary `&&` if lazy on the second argument. Meaning that if it meets `False` as the first argument, there is no need to calculate another arg. In `andM` we would like to keep the same behaviour, so the

```
Just False `andM` _
```

should be `Just False` :slightly_smiling_face: 


Note that with only `Applicative` you are not able to implement a lazier version of `andM`, you need to use `Monad` and not use `&&` inside as well 🙂

Reply:
Ok, will try to solve it again.